instrumentation/google.golang.org/grpc/otelgrpc/stats_handler.go

```diff
@@ -135,10 +135,14 @@ func (h *clientHandler) HandleConn(context.Context, stats.ConnStats) {
 
 func (c *config) handleRPC(ctx context.Context, rs stats.RPCStats, isServer bool) { // nolint: revive  // isServer is not a control flag.
 	span := trace.SpanFromContext(ctx)
-	gctx, _ := ctx.Value(gRPCContextKey{}).(*gRPCContext)
+	var metricAttrs []attribute.KeyValue
 	var messageId int64
-	metricAttrs := make([]attribute.KeyValue, 0, len(gctx.metricAttrs)+1)
-	metricAttrs = append(metricAttrs, gctx.metricAttrs...)
+
+	gctx, ok := ctx.Value(gRPCContextKey{}).(*gRPCContext)
+	if ok && gctx != nil {
+		metricAttrs = make([]attribute.KeyValue, 0, len(gctx.metricAttrs)+1)
```

Comment:
Why the extra one here?

Reply:
That is what was there to begin with.  Its to have space for the `rpcStatusAttr` if this is a `*stats.End` case.